src/test/java/org/itsallcode/openfasttrace/cli/ITestCliWithFilter.java

```diff
@@ -0,0 +1,146 @@
+package org.itsallcode.openfasttrace.cli;
+
+import static org.hamcrest.core.IsNot.not;
+
+/*-
+ * #%L
+ * OpenFastTrace
+ * %%
+ * Copyright (C) 2016 - 2018 itsallcode.org
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * #L%
+ */
+
+import static org.hamcrest.core.StringContains.containsString;
+import static org.junit.Assert.assertThat;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.itsallcode.openfasttrace.testutil.AbstractSpecificationFileBasedTest;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.Assertion;
+import org.junit.contrib.java.lang.system.ExpectedSystemExit;
+import org.junit.contrib.java.lang.system.SystemOutRule;
+import org.junit.rules.TemporaryFolder;
+
+public class ITestCliWithFilter extends AbstractSpecificationFileBasedTest
+{
+    public static final String SPECIFICATION = String.join(System.lineSeparator(), //
+            "`feat~a~1`", //
+            "I am a feature", //
+            "`req~b~2`", //
+            "A user requirement", //
+            "Tags: tag1", //
+            "`dsn~c~3`", //
+            "Design", //
+            "Tags: tag2", //
+            "`impl~d~4`");
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+    @Rule
+    public final SystemOutRule systemOutRule = new SystemOutRule().enableLog();
+    @Rule
+    public final ExpectedSystemExit exit = ExpectedSystemExit.none();
+
+    private File specFile;
+
+    @Before
+    public void before() throws IOException
+    {
+        this.specFile = this.tempFolder.newFile("spec.md");
+        writeTextFile(this.specFile, SPECIFICATION);
+        this.exit.expectSystemExitWithStatus(0);
+    }
+
+    // [itest->dsn~filtering-by-tags-during-import~1]
+    @Test
+    public void testWithoutFilter()
+    {
+        this.exit.checkAssertionAfterwards(new Assertion()
```

Comment:
Is it possible to use a lambda expression here? Then `ITestCliWithFilter.` should not be necessary.
If lambdas are not possible as a parameter: extract a method that accepts a lambda.

Reply:
Fixed: with pleasure. :-)